zhusuan/model/stochastic.py

```diff
@@ -571,3 +574,135 @@ def __init__(self,
+        dimensions in `batch_shape` (counted from the end) that are grouped
+        into a single event, so that their probabilities are calculated
+        together. Default is 0, which means a single value is an event.
+        See :class:`~zhusuan.distributions.base.Distribution` for more detailed
+        explanation.
+    :param is_reparameterized: A Bool. If True, gradients on samples from this
+        `StochasticTensor` are allowed to propagate into inputs, using the
+        reparametrization trick from (Kingma, 2013).
+    """
+
+    def __init__(self,
+                 name,
+                 temperature,
+                 logits,
+                 n_samples=None,
+                 group_event_ndims=0,
+                 is_reparameterized=True):
+        exp_concrete = distributions.ExpConcrete(
+            temperature,
+            logits,
+            group_event_ndims=group_event_ndims,
+            is_reparameterized=is_reparameterized
+        )
+        super(ExpConcrete, self).__init__(name, exp_concrete, n_samples)
+
+
+class Concrete(StochasticTensor):
+    """
+    The class of Concrete `StochasticTensor`.
+    See :class:`~zhusuan.model.base.StochasticTensor` for details.
+
+    :param temperature: A 0-D `float` Tensor. The temperature of the relaxed
+        distribution. The temperature should be positive.
+    :param logits: A N-D (N >= 1) `float` Tensor of shape (...,
+        n_categories). Each slice `[i, j, ..., k, :]` represents the
+        un-normalized log probabilities for all categories.
+
+        .. math:: \\mathrm{logits} \\propto \\log p
+
+    :param n_samples: A 0-D `int32` Tensor or None. Number of samples
+        generated by this `StochasticTensor`.
+    :param group_event_ndims: A 0-D `int32` Tensor representing the number of
+        dimensions in `batch_shape` (counted from the end) that are grouped
+        into a single event, so that their probabilities are calculated
+        together. Default is 0, which means a single value is an event.
+        See :class:`~zhusuan.distributions.base.Distribution` for more detailed
+        explanation.
+    :param is_reparameterized: A Bool. If True, gradients on samples from this
+        `StochasticTensor` are allowed to propagate into inputs, using the
+        reparametrization trick from (Kingma, 2013).
+    :param check_numerics: Bool. Whether to check numeric issues.
+    """
+
+    def __init__(self,
+                 name,
+                 temperature,
+                 logits,
+                 n_samples=None,
+                 group_event_ndims=0,
+                 is_reparameterized=True,
+                 check_numerics=False):
+        concrete = distributions.Concrete(
+            temperature,
+            logits,
+            group_event_ndims=group_event_ndims,
+            is_reparameterized=is_reparameterized,
+            check_numerics=check_numerics
+        )
+        super(Concrete, self).__init__(name, concrete, n_samples)
```

Comment:
Is it possible to have a way to automatically generate docstrings (even codes) for StochasticTensors based on distributions?